src/app/wishlists/wishlist-file.ts

```diff
@@ -85,6 +91,51 @@ export function toWishList(...fileTexts: string[]): WishListAndInfo {
   }
 }
 
+function autoUpgradeRoll(roll: WishListRoll): WishListRoll[] {
+  const allRolls: WishListRoll[] = [];
+
+  allRolls.push(roll);
+
+  roll.recommendedPerks.forEach((perk) => {
+    const enhancedRoll = getEnhancedRoll(roll, perk);
+
+    if (enhancedRoll) {
+      allRolls.push(enhancedRoll);
+
+      enhancedRoll.recommendedPerks.forEach((ePerk) => {
+        const innerEnhancedRoll = getEnhancedRoll(enhancedRoll, ePerk);
+
+        if (innerEnhancedRoll) {
+          allRolls.push(innerEnhancedRoll);
+        }
+      });
+    }
+  });
+
+  return allRolls;
+}
+
+function getEnhancedRoll(roll: WishListRoll, perkToCheck: number): WishListRoll | null {
+  const enhancedPerk = D2EnhancedTraits[perkToCheck];
+
+  if (enhancedPerk && !isNaN(enhancedPerk)) {
+    const cloneRoll = {
+      itemHash: roll.itemHash,
+      recommendedPerks: new Set(roll.recommendedPerks),
+      isExpertMode: roll.isExpertMode,
+      isUndesirable: roll.isUndesirable,
+      notes: roll.notes,
+    };
+
+    cloneRoll.recommendedPerks.delete(perkToCheck);
+    cloneRoll.recommendedPerks.add(enhancedPerk);
```

Comment:
This doesn't preserve the order of perks in the Armory view and it starts looking a bit confusing, but we could also change the Armory view to sort perks by socket index... 

![grafik](https://user-images.githubusercontent.com/14299449/159189402-2eff12f7-0c55-42db-99eb-10c7f174db39.png)

Reply:
Hmm yeah, probably the fix is on the Armory side.

Reply:
Thanks for poking at the Armory view (since I forgot to check that) - I hopefully didn't introduce any bugs with this changeset; having enhanced versions of perks available on craftable weapons I'm sure presents a weird design issue.